rust/pact_matching/src/lib.rs

```diff
@@ -1420,7 +1423,7 @@ pub fn match_message_metadata(
         },
         None => {
           result.insert(key.clone(), vec![Mismatch::MetadataMismatch { key: key.clone(),
-            expected: value.as_str().unwrap_or_default().to_string(),
+            expected: json_to_string(&value),
```

Comment:
@uglyog this look OK to you? (not introduced in this PR but the tests pass and my complex metadata check with Pact Go also passes)

Reply:
LGTM